### PR TITLE
Adds CI job to test strongSwan integration

### PR DESCRIPTION
**Pull request by geedo0**
### Issues:
Resolves PQCrypto-7

### Description of changes: 
Describe AWS-LC’s current behavior and how your code changes that behavior. If there are no issues this pr is resolving, explain why this change is necessary.

This CI job tests AWS-LC against the tip of strongswan/master. We use
AWS-LC's shared library build and override the libcrypto used by
strongSwan by setting LD_LIBRARY_PATH. We also run through strongSwan's
unit test framework to make sure there are no correctness issues.

### Call-outs:
Point out areas that need special attention or support during the review process. Discuss architecture or design changes.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

`act -W .github/workflows/integrations.yml -j strongswan`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/integration/run_strongswan_integration.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  export CFLAGS="-I${AWS_LC_INSTALL_FOLDER}/include"
+  export LDFLAGS="-L${AWS_LC_INSTALL_FOLDER}/lib"
+  autoreconf -ivf
+  # Mirror the configuration flags set by strongSwan's CI by taking all of the
+  # flags specified within CONFIG when TEST is openssl-awslc in their test.sh.
+  # https://github.com/strongswan/strongswan/blob/44e241fccc166211ccfdd322047c1213ff3ae73c/scripts/test.sh#L468
+  ./configure --disable-defaults --enable-pki --enable-openssl --enable-pem \
+  --disable-dependency-tracking --enable-silent-rules --enable-test-vectors \
+  --enable-monolithic=no --enable-leak-detective=no --enable-asan
+  make -j ${NUM_CPU_THREADS}
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_strongswan_integration.sh:32]: in other dynamically linked integration tests, we [use `ldd` to validate](https://github.com/aws/aws-lc/blob/main/tests/ci/integration/run_tcpdump_integration.sh#L40) that the project under test has correctly linked to _our_ libcrypto/libssl. can we do that here as well?

File: tests/ci/integration/run_strongswan_integration.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+function strongswan_run_tests() {
+  make -j ${NUM_CPU_THREADS} check
+}
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+git clone --depth 1 https://github.com/strongswan/strongswan.git ${STRONGSWAN_SRC_FOLDER}
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_strongswan_integration.sh:43]: looks like we're testing against their mainline, do we expect them to frequently make potentially breaking changes? if so, we might want to pin this to a particular version. this isn't a concern if we get an AWS-LC integration test into StrongSwan's CI.

File: tests/ci/integration/run_strongswan_integration.sh (hunk 1 above)
Comment (geedo0) [reply tests/ci/integration/run_strongswan_integration.sh:32]: Sure, that's straightforward.

File: tests/ci/integration/run_strongswan_integration.sh (hunk 2 above)
Comment (geedo0) [reply tests/ci/integration/run_strongswan_integration.sh:43]: I certainly don't expect them to do that, but maybe I'm naive? There will be an AWS-LC CI check added to strongSwan so perhaps we can be happy with a policy of mutually assured destruction?

Comment (WillChilds-Klein) [reply tests/ci/integration/run_strongswan_integration.sh:43]: > There will be an AWS-LC CI check added to strongSwan

nice, I think this minimizes the chance of any breaking changes getting merged downstream.

**Review by WillChilds-Klein — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Pull request merged**